Add letter frequency functionality

patrickmarabeas opened the pull request:
Implements two functions. One which creates a frequency map of given runes, the other sorts the map and returns a slice of "k:v" strings

Fixes #418 

Review of colleague's PR #527

runnerdave commented on 03_letters/patrickmarabeas/main.go:
```diff
@@ -0,0 +1,53 @@
+	return result
+}
+
+// Sorted letter kvp slice
```
what is kvp?

patrickmarabeas replied:
key value pair

runnerdave commented on 03_letters/patrickmarabeas/main.go:
```diff
@@ -0,0 +1,53 @@
+		i++
+	}
+	// Sort the slice of runes
+	// The less function that sort implements takes int32 which aligns with rune
```
I believe these comments are unnecessary, the code should be self explanatory

patrickmarabeas replied:
As a Go scrub, it really isn't. Initially I was using casting to solve this - but that felt wrong. This documents *for myself* how I got around the need to do so, and why this works (how the types align).

runnerdave commented on 03_letters/patrickmarabeas/main_test.go:
```diff
@@ -0,0 +1,82 @@
+		input    string
+		expected map[rune]int
+	}{
+		"Letters": {
```
you may consider testing more complex combinations such as uppercases and longer strings, as well as empty strings

patrickmarabeas replied:
Some additional tests added.

runnerdave reviewed (changes requested):
minor changes, but the main important is your PR description, I believe you should review other person's PR and reference it there, as well as listing your changes in the description

runnerdave reviewed (changes requested):
Hi, saw you addressed the extra test, but you also need to improve your PR description

patrickmarabeas commented:
PR has been updated with reviewed #

patrickmarabeas committed:
Add letter frequency functionality

Implements two functions. One which creates a frequency map of given
runes, the other sorts the map by ascending key and returns a slice of
"k:v" strings

Closes Lab 03 (#418)

patrickmarabeas commented:
Updated doc string

pentaphobe reviewed (approved):
LGTM!

runnerdave reviewed (approved):
👍

patrickmarabeas committed:
Merge branch 'master' into lab-03-pm
